server/src/com/thoughtworks/go/server/controller/PipelineHistoryController.java

```diff
@@ -106,7 +108,10 @@ public ModelAndView list(@RequestParam("pipelineName")String pipelineName,
 
         PipelinePauseInfo pauseInfo = pipelinePauseService.pipelinePauseInfo(pipelineName);
         boolean hasBuildCauseInBuffer = pipelineScheduleQueue.hasBuildCause(CaseInsensitiveString.str(pipelineConfig.name()));
-        PipelineInstanceModels pipelineHistory = pipelineHistoryService.load(pipelineName, pagination, username, true);
+        PipelineInstanceModels pipelineHistory = (labelFilter == null || labelFilter.trim().isEmpty()) ?
```

Comment:
Sorry, missed this earlier.
We have a utility method https://github.com/gocd/gocd/blob/master/base/src/com/thoughtworks/go/util/StringUtil.java#L161 for checking blank strings can you use that here?